### PR TITLE
RFC 54: Quilt Kotlin Team

### PR DESCRIPTION
**Pull request by Oliver-makes-code**
This RFC aims to define the Quilt Kotlin Team, and its uses

### CODE REVIEW

**Comment by Oliver-makes-code** [comment]
So, for the Drawbacks part, I left it as "TBD" because we can find drawbacks for it through discussion. There's also the unresolved questions, the first of which, "Should this be a subdivision QSL?"

**Comment by Oliver-makes-code** [comment]
> QKL shouldn't be a subdivision of QSL, as QSL is based on Java, it'd be
difficult for maintainers who don't know Kotlin to be able to maintain QKL.
Having QKL as a subdivision of QSL would mean that changes in the respective
QSL modules would have to be copied to the QKL module before merging. This is a
bad idea, because it forces everyone to know Kotlin to be able to contribute.

File: structure/0054-quilt-kotlin-team.md
Comment (spiceswag) [inline structure/0054-quilt-kotlin-team.md:40]: FLK also adds support for the main class of a mod to be a Kotlin object

File: structure/0054-quilt-kotlin-team.md
Comment (spiceswag) [inline structure/0054-quilt-kotlin-team.md:41]: One of the main selling points of kotlin is the fact that it can interop almost flawlessly with java code meaning that there is not much room for improvement here.

Comment (Oliver-makes-code) [reply structure/0054-quilt-kotlin-team.md:41]: Interop doesn't necessarily mean ideal

**Comment by Oliver-makes-code** [comment]
Kotlin's interoperability with Java is just that, interoperability. There are some ideas that work well in Java, but not Kotlin, and vice versa. Extension functions and DSLs don't exist in Java, and with this we'd provide essential ones for modding in Kotlin. Also once CHASM is added to loader, we can provide an official front end for it in Kotlin

File: structure/0054-quilt-kotlin-team.md
Comment (P03W) [inline structure/0054-quilt-kotlin-team.md:5]: Do we want to mention possible other projects? Such as PRing build tools support or a kotlin chasm frontend?

File: structure/0054-quilt-kotlin-team.md
Comment (P03W) [inline structure/0054-quilt-kotlin-team.md:34]: Technically you could request help for the contribution, so imo it's more a problem of the general pool of kotlin users available and willing to help, basically creating a low bus factor in these situations unnecessarily

File: structure/0054-quilt-kotlin-team.md
Comment (P03W) [inline structure/0054-quilt-kotlin-team.md:51]: Iirc this was almost unanimously yes for things like brigadier

Unless you mean 3rd party libs like CCA and whatever?

**Review by P03W — CHANGES_REQUESTED** [review]

**Comment by Southpaw1496** [comment]
Couple of things:
- Firstly, if this team's sole purpose is making QKL, shouldn't they be the QKL team, just like the people making QSL are the QSL team? I understand there has been discussion about building Kotlin wrappers for things outside QSL (which I am all for), but that's been put under "Unresolved Questions".
- Secondly, if QKL will be for QSL libraries, wouldn't that be something like Quilt Standard Libraries Kotlin, or Quilt Standard Kotlin Libraries, not Quilt Kotlin Libraries? The name is inconsistent with QSL (it's not Quilt Java Libraries, after all).

File: structure/0054-quilt-kotlin-team.md
Comment (Oliver-makes-code) [reply structure/0054-quilt-kotlin-team.md:51]: I mean 3rd party libraries

File: structure/0054-quilt-kotlin-team.md
Comment (Oliver-makes-code) [reply structure/0054-quilt-kotlin-team.md:5]: I can do that

**Comment by Oliver-makes-code** [comment]
> - Firstly, if this team's sole purpose is making QKL, shouldn't they be the QKL team, just like the people making QSL are the QSL team? I understand there has been discussion about building Kotlin wrappers for things outside QSL (which I am all for), but that's been put under "Unresolved Questions".

The main reason is that Quilt Kotlin Team rolls off the tongue better, but it can be changed to QKL team if need be

> - Secondly, if QKL will be for QSL libraries, wouldn't that be something like Quilt Standard Libraries Kotlin, or Quilt Standard Kotlin Libraries, not Quilt Kotlin Libraries? The name is inconsistent with QSL (it's not Quilt Java Libraries, after all).

QKL will be separate from QSL

**Comment by Southpaw1496** [comment]
> QKL will be separate from QSL
I know, but since you're porting the standard libraries, it makes sense to say as much in the name.

**Comment by Oliver-makes-code** [comment]
We're not porting the standard libraries, per se, we're wrapping the libraries to make it feel more familiar to kotlin developers

File: structure/0054-quilt-kotlin-team.md
Comment (sschr15) [inline structure/0054-quilt-kotlin-team.md:52]: Would https://github.com/sschr15/khasm be a reasonable module to port (likely with a rename) into a Chasm frontend?

Comment (Oliver-makes-code) [reply structure/0054-quilt-kotlin-team.md:52]: More than likely, yes

**Comment by Oliver-makes-code** [comment]
All that's still TBD is the Drawbacks section.

File: structure/0054-quilt-kotlin-team.md
Comment (Oliver-makes-code) [reply structure/0054-quilt-kotlin-team.md:34]: > Technically you could request help for the contribution, so imo it's more a problem of the general pool of kotlin users available and willing to help, basically creating a low bus factor in these situations unnecessarily

That's the idea I was trying to get across

**Comment by P03W** [comment]
The main drawback I can think of is that QKL may get out of date with QSL and/or the versioning scheme may cause confusion with QSL versioning (i.e QSL updates to 1.0.1 but QKL sticks on 1.0.0 as its not breaking) although that could be mitigated somewhat with minimum QSL version requirements

**Comment by Oliver-makes-code** [comment]
Open for review \o/

**Review by Southpaw1496 — APPROVED** [review]
I'm still not too hot on the name "QKL" (I think QSLK would make more sense), that is a very minor complaint and other than that it seems like an excellent idea.

**Review by NoComment1105 — APPROVED** [review]
Looks good

**Comment by Oliver-makes-code** [comment]
> I think QSLK would make more sense
@Southpaw1496 
The main reason I'm not going with QSLK is due to the fact it would cause people to presume it's a subdivision of QSL, which it's not

**Review by sschr15 — APPROVED** [review]

**Comment by CheaterCodes** [comment]
In accordance with the (not yet implemented) Governance Amendment #47, I would like to see you propose initial team members and the initial team lead. This would not be part of the RFC, but should at least be mentioned in a comment here so that people can approve of it.

**Comment by Oliver-makes-code** [comment]
> In accordance with the (not yet implemented) Governance Amendment #47, I would like to see you propose initial team members and the initial team lead. This would not be part of the RFC, but should at least be mentioned in a comment here so that people can approve of it.

I can be the initial team lead, and @NoComment1105, @P03W, and I can be the initial members. @gdude2002 said he'd consider joining at a later date

**Comment by sschr15** [comment]
I'd like to be an initial member as well

**Review by PotatoPresident — APPROVED** [review]

**Comment by PotatoPresident** [comment]
Definitely would like to be a member

**Review by TibiNonEst — APPROVED** [review]

**Review by leocth — APPROVED** [review]
:+1: I'd like to be in the team as well

**Comment by chaottic** [comment]
I also would like to be in the team. 😊

**Comment by gdude2002** [comment]
I think it's necessary to point out that you can contribute without being on the team - it should not be a large team, and the people on it should be there to steer the projects that the team is responsible for, not just because they want to help with the projects.

**Review by Haven-King — APPROVED** [review]

**Review by mrmangohands — APPROVED** [review]

**Comment by Oliver-makes-code** [comment]
Now that this has approval from all 3 admins this should be able to enter FCP

**Pull request merged**
